src/StrawberryShake/Client/src/Transport.WebSockets/Protocols/GraphQlWs/SubscriptionTransportWsProtocol.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using StrawberryShake.Http.Subscriptions;
+using StrawberryShake.Transport.Http;
+
+namespace StrawberryShake.Transport.WebSockets
+{
+    public class SubscriptionTransportWsProtocol : ISocketProtocol
```

Comment:
This is a implementation of the graphql-ws socket protocol. 

The MessagePipelineHandler is probably also  protocol specific